### ISSUE TITLE
Disable gitsigns/change background for file list window

### ISSUE THREAD

**Issue by gegoune** [sindrets/diffview.nvim#143]
Hi, amazing plugin, thanks a lot!

I've got two questions:

1) Is it common to disable gitsigns in diffview windows? It feels bit redundant to have those signs in there too. What do you think? How would one disable gitsigns for diffs showed by your plugin?

2) I couldn't find a highlight group for window showing file list - I would like to dim background on it to make it match nvim-tree.

Thanks!

**Comment by sindrets** [comment]
> Hi, amazing plugin, thanks a lot!

@gegoune Thank you so much!

> Is it common to disable gitsigns in diffview windows? It feels bit redundant to have those signs in there too. What do you think? How would one disable gitsigns for diffs showed by your plugin?

You can probably use the `:h diffview-config-hooks` for this. You could call `:Gitsigns detach` on `diff_buf_win_enter`:

```lua
hooks = {
  diff_buf_win_enter = function(bufnr)
    require("gitsigns").detach(bufnr)
  end,
}
```

The problem with this is that the buffer is now detached outside of diffview as well. I suppose I could add hooks for `view_enter`, and `view_leave`. That way you can call `gitsigns.toggle_signs({boolean})` on those events.

Another approach would be to just disable the signcolumn if you don't need it in diffviews:

```lua
hooks = {
  diff_buf_win_enter = function(bufnr)
    vim.opt_local.signcolumn = "no"
  end,
}
```

> I couldn't find a highlight group for window showing file list - I would like to dim background on it to make it match nvim-tree.

The highlight group you're looking for is `DiffviewNomal`. This is used in all the panels I believe.

**Comment by gegoune** [comment]
> The problem with this is that the buffer is now detached outside of diffview as well. I suppose I could add hooks for `view_enter`, and `view_leave`.

Do you think it would be beneficial to the plugin? If so I would definitely use it to toggle gitsigns.

> Another approach would be to just disable the signcolumn if you don't need it in diffviews:
> 
> ```lua
> hooks = {
>   diff_buf_win_enter = function(bufnr)
>     vim.opt_local.signcolumn = "no"
>   end,
> }
> ```

Done and works just fine. I am not sure if there is anything that I might be missing on by disabling `signcolumn` but unless aforementioned hooks are an option that will have to do. Thanks for that hint!


> > I couldn't find a highlight group for window showing file list - I would like to dim background on it to make it match nvim-tree.
> 
> The highlight group you're looking for is `DiffviewNormal`. This is used in all the panels I believe.

I have added that group to nightfox's configuration but it does not seem to do anything. Is loading order important here? Does diffview initialise at start requiring groups to already exist?

**Comment by sindrets** [comment]
> Do you think it would be beneficial to the plugin? If so I would definitely use it to toggle gitsigns.

Yes, these kinds of hooks are good for extensibility, and trivial to implement. I'll add them soon.

> I have added that group to nightfox's configuration but it does not seem to do anything. Is loading order important here? Does diffview initialise at start requiring groups to already exist?

I'm not familiar with nightfox, but in my own config I have a `apply_tweaks()` function where I apply highlight overrides, that I trigger on the `ColorScheme` autocommand event. This works regardless of the color scheme (with a few exceptions, due to some color schemes making use of `vim.schedule()` causing highlights to be applied out of order). This definitely works for diffview though. I only use `:hi default` in the plugin, so the plugin does not override highlights if you set them explicitly.

**Comment by gegoune** [comment]
> > Do you think it would be beneficial to the plugin? If so I would definitely use it to toggle gitsigns.
> 
> Yes, these kinds of hooks are good for extensibility, and trivial to implement. I'll add them soon.

Thank you!

> I only use `:hi default` in the plugin, so the plugin does not override highlights if you set them explicitly.
Ah, apologies, I had a typo which prevented it from working, feeling so stupid now. Apologies for wasting your time and thanks for that highlight group name.

Should I close this issue or wait for new hooks to arrive?

Huge thanks once again!

**Comment by sindrets** [comment]
You can leave it open. I'll auto-close it in a commit with the hooks :)

Just wanted to note: I checked out nightfox, and to my surprise I found that they are actually using a modified version of [my own color library](https://github.com/sindrets/dotfiles/blob/master/.config/nvim/lua/nvim-config/common/color.lua) that I made for my own config :smile:
